seq_tools/validation/c160_file_r1_r2_check.py

```diff
@@ -0,0 +1,118 @@
+            if 'is_paired_end' not in rg:
+                message = "Required field 'is_paired_end' is not found in metadata JSON"
+                self.logger.info(message)
+                self.message = message
+                self.status = 'INVALID'
+                return
+
+            if not isinstance(rg['is_paired_end'], bool):
+                message = "Required field 'is_paired_end' should be Boolean type."
+                self.logger.info(message)
+                self.message = message
+                self.status = 'INVALID'
+                return
+            
+            if rg['is_paired_end']:
+                if not 'file_r1' in rg or not rg['file_r1']:
+                    message = "Required field 'file_r1' is not found in metadata JSON for paired end sequencing reads."
+                    self.logger.info(message)
+                    self.message = message
+                    self.status = 'INVALID'
+                    return
+                if not 'file_r2' in rg or not rg['file_r2']:
+                    message = "Required field 'file_r2' is not found in metadata JSON for paired end sequencing reads."
+                    self.logger.info(message)
+                    self.message = message
+                    self.status = 'INVALID'
+                    return
+                if rg['file_r1'].endswith('.bam') and not rg['file_r1'] == rg['file_r2']:
+                    message = "Fields 'file_r1' and 'file_r2' should be the same for paired end BAM sequencing reads."
+                    self.logger.info(message)
+                    self.message = message
+                    self.status = 'INVALID'
+                if not rg['file_r1'].endswith('.bam') and rg['file_r1'] == rg['file_r2']:
+                    message = "Fields 'file_r1' and 'file_r2' should NOT be the same for paired end FASTQ sequencing reads."
+                    self.logger.info(message)
+                    self.message = message
+                    self.status = 'INVALID'
+                
+            else:
+                if 'file_r2' in rg and rg['file_r2']:
+                    message = "Field 'file_r2' should NOT been populated in metadata JSON for single end sequencing reads."
+                    self.logger.info(message)
+                    self.message = message
+                    self.status = 'INVALID'
+                    return 
+
+            if rg.get('file_r1'): fns.add(rg['file_r1'])
+            if rg.get('file_r2'): fns.add(rg['file_r2'])
+
+        fls = set()
+        for fl in self.metadata.get('files'):
+            if not 'fileName' in fl or not fl['fileName']:
+                message = "Required field 'fileName' is not found in metadata JSON."
+                self.logger.info(message)
+                self.message = message
+                self.status = 'INVALID'
+                return
+            
+            fls.add(fl.get('fileName'))
+
+        if not fns == fls:
```

Comment:
I'd suggest to break this into two different scenarios:
```
if fns - fls:
  missing_files = fns  - fls
  message = "File(s) specified in 'file_r1' or 'file_r2' missed in 'files' section of the metadata JSON: %s" % ", ".join(missing_files)
  # ...
elif fls - fns:
  extra_files = fls  - fns
  message = "Extra file(s) found in 'files' section but not used in 'file_r1' or 'file_r2' in the metadata JSON: %s" % ", ".join(missing_files)
  # ...
```

Reply:
Instead of using different check, can we use this to cover https://github.com/icgc-argo/seq-tools/issues/21 as well?

Reply:
good point, I forgot about `c190` in #21. as we are trying to keep each checker simple, maybe should move this port here to `c190`. basically this checker focuses on files `file_r1` / `file_r2` must exist in `files`. #21 focuses on files in `files` section must exist in `file_r1` / `file_r2`.